uncaughtException TypeError: Cannot read property 'command' of undefined

Hello,
How are you?
I experience an error such as:
```text
May 10 10:55:01 server p3xrs.js[21488]: [P3XRS] 2019/05/10 10:55:01.177 [ERROR]  [PID: 021488] 5/10/2019, 10:55:01 AM uncaughtException TypeError: Cannot read property 'command' of undefined
May 10 10:55:01 server p3xrs.js[21488]:     at Redis.exports.returnError (/var/p3x-redis-ui-server/node_modules/ioredis/built/redis/parser.js:36:20)
May 10 10:55:01 server p3xrs.js[21488]:     at JavascriptRedisParser.returnError (/var/p3x-redis-ui-server/node_modules/ioredis/built/redis/parser.js:20:19)
May 10 10:55:01 server p3xrs.js[21488]:     at JavascriptRedisParser.execute (/var/p3x-redis-ui-server/node_modules/redis-parser/lib/parser.js:542:14)
May 10 10:55:01 server p3xrs.js[21488]:     at Socket.<anonymous> (/var/p3x-redis-ui-server/node_modules/ioredis/built/redis/event_handler.js:107:26)
May 10 10:55:01 server p3xrs.js[21488]:     at Socket.emit (events.js:193:13)
May 10 10:55:01 server p3xrs.js[21488]:     at addChunk (_stream_readable.js:295:12)
May 10 10:55:01 server p3xrs.js[21488]:     at readableAddChunk (_stream_readable.js:276:11)
May 10 10:55:01 server p3xrs.js[21488]:     at Socket.Readable.push (_stream_readable.js:231:10)
May 10 10:55:01 server p3xrs.js[21488]:     at TCP.onStreamRead (internal/stream_base_commons.js:150:17)
```
https://github.com/patrikx3/redis-ui/issues/14

--- a/lib/command.ts
+++ b/lib/command.ts
@@ -220,7 +220,7 @@ export default class Command {
     }
 
     let result
-    let commandStr = '*' + (this.args.length + 1) + '\r\n$' + this.name.length + '\r\n' + this.name + '\r\n'
+    let commandStr = '*' + (this.args.length + 1) + '\r\n$' + Buffer.byteLength(this.name) + '\r\n' + this.name + '\r\n'
     if (bufferMode) {
       const buffers = new MixedBuffers();
       buffers.push(commandStr);

--- a/test/functional/send_command.js
+++ b/test/functional/send_command.js
@@ -217,6 +217,7 @@ describe('send command', function () {
     } catch(e) {
       err = e.message
     }
-    expect(err).to.contain(`unknown command \`${invalidCommand}\``)
+    expect(err).to.contain('unknown command')
+    expect(err).to.contain(invalidCommand)
   })
 });

--- a/test/functional/send_command.js
+++ b/test/functional/send_command.js
@@ -207,4 +207,16 @@ describe('send command', function () {
       });
     });
   });
+
+  it('throws for invalid command', async () => {
+    const redis = new Redis()
+    const invalidCommand = 'áéűáű'
+    let err
+    try {
+      await redis.call(invalidCommand, [])
+    } catch(e) {
+      err = e.message
+    }
+    expect(err).to.contain(`unknown command \`${invalidCommand}\``)
+  })
 });

